sui/deploy-contract.js

```diff
@@ -23,49 +54,179 @@ async function processCommand(contractName, config, chain, options) {
+
+            contractConfig.objects = {
+                gateway,
+                relayerDiscovery,
+                upgradeCap,
+            };
+            contractConfig.domainSeparator = domainSeparator;
+            contractConfig.operator = operator;
+            contractConfig.minimumRotationDelay = minimumRotationDelay;
+            break;
+        }
+
+        default: {
             throw new Error(`${contractName} is not supported.`);
+        }
     }
 
     printInfo(`${contractName} deployed`, JSON.stringify(contractConfig, null, 2));
 }
 
-async function mainProcessor(contractName, options, processor) {
+async function upgrade(contractName, policy, config, chain, options) {
+    const [keypair, client] = getWallet(chain, options);
+    const { packageDependencies } = options;
+    options.policy = policy;
+
+    await printWalletInfo(keypair, client, chain, options);
+
+    if (!chain.contracts[contractName]) {
+        throw new Error(`Cannot find specified contract: ${contractName}`);
+    }
+
+    const contractsConfig = chain.contracts;
+    const packageConfig = contractsConfig?.[contractName];
+
+    validateParameters({ isNonEmptyString: { contractName } });
+
+    if (packageDependencies) {
+        for (const dependencies of packageDependencies) {
+            const packageId = contractsConfig[dependencies]?.address;
+            updateMoveToml(dependencies, packageId);
+        }
+    }
+
+    const builder = new TxBuilder(client);
+    await upgradePackage(client, keypair, contractName, packageConfig, builder, options);
+}
+
+async function mainProcessor(args, options, processor) {
     const config = loadSuiConfig(options.env);
-    await processor(contractName, config, config.sui, options);
+    await processor(...args, config, config.sui, options);
     saveConfig(config, options.env);
+
+    if (options.offline) {
+        const { txFilePath } = options;
+        validateParameters({ isNonEmptyString: { txFilePath } });
+
+        const txB64Bytes = toB64(options.txBytes);
+
+        writeJSON({ message: options.offlineMessage, status: 'PENDING', unsignedTx: txB64Bytes }, txFilePath);
+        printInfo(`Unsigned transaction`, txFilePath);
+    }
 }
 
 if (require.main === module) {
     const program = new Command();
 
-    program
-        .name('deploy-contract')
-        .addOption(new Option('--packageName <packageName>', 'Package name to deploy'))
-        .addArgument(new Argument('<contractName>', 'Contract name to deploy').choices(Object.keys(contractMap)))
-        .description('Deploy SUI modules');
-
-    addBaseOptions(program);
-
-    program.action((contractName, options) => {
-        mainProcessor(contractName, options, processCommand);
-    });
+    program.name('deploy-contract').description('Deploy/Upgrade packages');
+
+    const deployCMD = program
+        .name('deploy')
+        .description('Deploy SUI modules')
```

Comment:
```suggestion
        .description('Deploy a Sui package')
```
Modules exist within a package